tensorboard/uploader/exporter.py

```diff
@@ -180,19 +223,125 @@ def _request_scalar_data(self, experiment_id, read_time):
             metadata = base64.b64encode(
                 response.tag_metadata.SerializeToString()
             ).decode("ascii")
-            wall_times = [
-                t.ToNanoseconds() / 1e9 for t in response.points.wall_times
-            ]
-            yield {
+            json_data = {
                 u"run": response.run_name,
                 u"tag": response.tag_name,
                 u"summary_metadata": metadata,
-                u"points": {
-                    u"steps": list(response.points.steps),
-                    u"wall_times": wall_times,
-                    u"values": list(response.points.values),
-                },
             }
+            filename = None
+            if response.HasField("points"):
+                json_data[u"points"] = self._process_scalar_points(
+                    response.points
+                )
+                filename = _FILENAME_SCALARS
+            elif response.HasField("blob_sequences"):
+                json_data[u"points"] = self._process_blob_sequence_points(
+                    response.blob_sequences, experiment_id
+                )
+                filename = _FILENAME_BLOB_SEQUENCES
+            if filename:
+                yield json_data, filename
+            else:
+                logger.warn(
+                    "Skipping a response from experiment-data stream "
+                    "due to the lack of any valid data type (such as "
+                    "scalars and blob sequences): run=%s, tag=%s"
+                    % (response.run_name, response.tag_name)
```

Comment:
I prefer to keep this as a warning. If someone gets this warning after we roll out tensor support, it means their uploader is out of date, so it is actionable in that sense. I added a sentence in this warning to indicate that.
This is also a somewhat rare occurrence, because if a user has uploaded tensor data before, they should most likely already installed a new version.

Reply:
Okay, fine with me with the new sentence; thanks!